src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -394,5 +394,146 @@ public static void EnumAsSByteMax()
         }
+
+        [Fact]
+        public static void ToString_ZeroEnum_ReturnsZero()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleSByteEnum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleInt64Enum)0);
+            Assert.Equal("0", json);
+        }
+
+        [Fact]
+        public static void ToString_MaxValue_ReturnsMaxValue()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)byte.MaxValue);
+            Assert.Equal(byte.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleSByteEnum)sbyte.MaxValue);
+            Assert.Equal(sbyte.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)UInt32.MaxValue);
+            Assert.Equal(UInt32.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)UInt64.MaxValue);
+            Assert.Equal(UInt64.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleInt64Enum)long.MaxValue);
+            Assert.Equal(long.MaxValue.ToString(), json);
+        }
+
+        [Fact]
+        public static void ToString_MinValue_ReturnsMinValue()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)byte.MinValue);
+            Assert.Equal(byte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleSByteEnum)sbyte.MinValue);
+            Assert.Equal(sbyte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)ushort.MinValue);
+            Assert.Equal(ushort.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)UInt32.MinValue);
+            Assert.Equal(UInt32.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)UInt64.MinValue);
+            Assert.Equal(UInt64.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleInt64Enum)long.MinValue);
+            Assert.Equal(long.MinValue.ToString(), json);
+        }
+
+        [Theory]
+        [InlineData(byte.MaxValue + 1, (SampleByteEnum)0)]
+        [InlineData(byte.MaxValue + 2, (SampleByteEnum)1)]
+        [InlineData(byte.MaxValue + 13, (SampleByteEnum)12)]
+        [InlineData(byte.MaxValue * 2, (SampleByteEnum)byte.MaxValue - 1)]
+        public static void ToString_Byte(ulong i, SampleByteEnum enumValue)
```

Comment:
nit: Maybe it would be simpler for this to be the following: `string expectedString, SampleByteEnum enumValue`

It would be easier to see what's happening. Currently, the person debugging would have to reason about the subtraction and modulo operations. I think adding a comment to explain how we came about the `expectedString` value would be sufficient.